File: LAB_1/LAB_1_Csharp/LAB_1_Csharp/Counter.cs
using System;

namespace LAB_1_Csharp
{
    public class Counter : IComparable<Counter>
    {
        private int _id;
        private long _step;
        public int TimeOfWorking { get; set; }
        public bool CanStop { get; set; }

        public Counter(int id, long step, int timeOfWork)
        {
            _id = id;
            _step = step;
            TimeOfWorking = timeOfWork;
            CanStop = false;
        }
        public int CompareTo(Counter other)
        {
            return TimeOfWorking.CompareTo(other.TimeOfWorking);
        }

        public void CounterOfSumAndItems()
        {
            long sumOfSequence = 0;
            long itemsUsed = 0;

            do
            {
                sumOfSequence += _step;
                itemsUsed++;

            } while (!CanStop);

            Console.WriteLine($"ID: {_id} || sumOfSequence: {sumOfSequence} || itemsUsed: {itemsUsed}");
        }

        public void Switch()
        {
            CanStop = true;
        }
    }
}

File: LAB_1/LAB_1_Java/src/Counter.java
public class Counter extends Thread implements Comparable<Counter>
{
    private Integer _id;
    private Long _step;
    public Integer _timeOfWorking;
    public Integer getTimeOfWork()
    {
        return _timeOfWorking;
    }
    private  Boolean _canStop;

    public Counter(Integer id, Long step, Integer timeOfWorking)
    {
        _id = id;
        _step = step;
        _timeOfWorking = timeOfWorking;
        _canStop = false;
    }

    @Override
    public int compareTo(Counter o)
    {
        return _timeOfWorking.compareTo(o._timeOfWorking);
    }

    @Override
    public void run()
    {
        Long sumOfSequence = 0l;
        Long itemsUsed = 0l;

        do {
            sumOfSequence += _step;
            itemsUsed++;
        }while (!_canStop);

        System.out.println("ID: " + _id + " || sumOfSequence: " + sumOfSequence + " || itemsUsed: " + itemsUsed);
    }

    public void Switch()
    {
        _canStop = true;
    }
}
